public/components/FeatureAnywhereContextMenu/AddAlertingMonitor/AddAlertingMonitor.tsx

```diff
@@ -0,0 +1,249 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useMemo, useState } from 'react';
+import {
+  EuiFlexItem,
+  EuiFlexGroup,
+  EuiFlyoutHeader,
+  EuiFlyoutBody,
+  EuiFlyoutFooter,
+  EuiTitle,
+  EuiSpacer,
+  EuiButton,
+  EuiButtonEmpty,
+  EuiFormFieldset,
+  EuiCheckableCard,
+} from '@elastic/eui';
+import './styles.scss';
+import CreateNew from './CreateNew';
+import AssociateExisting from './AssociateExisting';
+import _ from 'lodash';
+import { Formik } from 'formik';
+import {
+  getInitialValues,
+  submit,
+} from '../../../pages/CreateMonitor/containers/CreateMonitor/utils/helpers';
+import {
+  createSavedObjectAssociation,
+  validateAssociationIsAllow,
+} from '../../../utils/savedObjectHelper';
+import { SEARCH_TYPE } from '../../../utils/constants';
+import { getClient, getNotifications } from '../../../services';
+
+function AddAlertingMonitor({
+  embeddable,
+  closeFlyout,
+  flyoutMode,
+  setFlyoutMode,
+  monitors,
+  selectedMonitorId,
+  setSelectedMonitorId,
+  index,
+  detectorId,
+  isAssociateAllowed,
+  limitReachedCallout,
+}) {
+  const history = {
+    location: { pathname: '/create-monitor', search: '', hash: '', state: undefined },
+    push: () => null,
+    goBack: closeFlyout,
+  };
+  const setFlyout = () => null;
+  const httpClient = getClient().httpClient;
+  const notifications = getNotifications();
+  const title = embeddable.vis.title;
+  const timeField = embeddable.vis?.data?.aggs?.aggs?.[1]?.params?.field?.displayName;
```

Comment:
Instead of these many checks, I would say, let's use the `_.get(...)` method from lodash
Also, if vis is expected to be null, then we should add that check in line 57 as well